srp_daemon/srp_ib_types.h

```diff
@@ -358,15 +340,15 @@ typedef struct _ib_mad_notice_attr		// Total Size calc  Accumulated
 			__be32		qp1;		// 4
 			uint8_t		qp2_msb;	// 1
 			__be16		qp2_lsb;	// 2
-			ib_gid_t	gid1;		// 16
-			ib_gid_t	gid2;		// 16
+			union ibv_gid	gid1;		// 16
+			union ibv_gid	gid2;		// 16
 			__be16		sw_lid;		// 2
 			uint8_t		port_no;	// 1
 		} PACK_SUFFIX ntc_259;
```

Comment:
This layout prior to change does not match IB spec for this trap. Is it used by SRP ?

In the spec, I don't think the GIDs are 8 byte aligned.